src/stylized/FacebookStyle.js

```diff
@@ -1,10 +1,10 @@
 //@flow
 import * as React from 'react'
-import Wrap from '../Wrap'
+import Wrap, { defaultProps } from '../Wrap'
 import type { WrapProps } from '../Wrap'
 
 const FacebookStyle = (props: WrapProps): React.Element<*> => (
-  <Wrap {...props}>
+  <Wrap {...defaultProps} {...props} >
```

Comment:
What is the advantage of doing this merge `{...defaultProps} {...props}` here instead of having `FacebookStyle.defaultProps = defaultProps` below the function component declaration ?

Reply:
I believe there is no perf difference, but this form in case of SFC is somewhat 'better' (but really I just have thought about it just today when doing this refactoring) because a static property assignment (such as `FacebookStyle.defaultProps = defaultProps`) is treated by UglifyJS as side effect - that means it wont be able to remove it, even if it recognizes `FacebookStyle` to be unused and therefore 'dead code'. 

Look at this input:
```js
var Test = function () {};
```
Output from uglifyJS
```js
```

Input:
```js
var Test = function () {};
Test.default = test;
```
Output from uglifyJS
```js
(function(){}).default=test;
```

Note - I've used such command on the sample file `uglifyjs -mc passes=3,unsafe --toplevel`

I advise strongly against using static properties whatsoever, unless ofc you have a reason to. Ofc its often not possible to not use static properties when dealing with React component - but at least the situation will get better for class components in upcoming babel7 (it will create static properties in a closure, making it more tree-shakeable). However problem for SFC will preserve - unless you do some "weirdo" hacks like:
```js
const MySFC = /*#__PURE__*/ (() => {
  function MySFC(props) {
    return <div/>
  }
  MySFC.propTypes = {}
  return MySFC
})()
```

or 
```js
function MySFC(props) {
  MySFC = function MySFC(props) {
    return <div/>
  }
  MySFC.propTypes = {}
  return MySFC(props)
}
```

Reply:
Actually latter wouldnt work for static properties case, as they wont be there until u make a first call to `MySFC`. The trick is "nice" for other "tricky" cases where minifiers are having hard times to do, in example consider this:
```js
function _extends() {
  _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

  return _extends.apply(this, arguments);
}
```